File: backend/src/main/kotlin/com/tkppp/sportresult/kbo/domain/KboRank.kt
package com.tkppp.sportresult.kbo.domain

import com.tkppp.sportresult.kbo.util.KboTeam
import javax.persistence.*

@Entity
class KboRank(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    val rank: Int,

    @Enumerated(EnumType.STRING)
    var name: KboTeam,

    @Column(nullable = false)
    var played: Int,

    @Column(nullable = false)
    var win: Int,

    @Column(nullable = false)
    var draw: Int,

    @Column(nullable = false)
    var defeat: Int,

    @Column(nullable = false)
    var winRate: Float,

    @Column(nullable = false)
    var gameDiff: Float,
)

File: node-backend/src/domain/kbo/kboRank.ts
import { Column, Entity, PrimaryGeneratedColumn, BaseEntity } from 'typeorm';
import { Team, teamEnum } from './kbo.utils'

@Entity()
export class KboRank extends BaseEntity {
  @PrimaryGeneratedColumn()
  id: number

  @Column({ nullable: false })
  ranking: number

  @Column({ nullable: false, type: 'enum', enum: teamEnum })
  name: Team

  @Column({ nullable: false })
  played: number

  @Column({ nullable: false })
  win: number

  @Column({ nullable: false })
  draw: number

  @Column({ nullable: false })
  defeat: number

  @Column({ nullable: false, type: 'float'})
  winRate: number

  @Column({ nullable: false, type: 'float'})
  gameDiff: number
}
